### PR TITLE
Integrate jtreceval

### PR DESCRIPTION
**Pull request by jasper-xian**
WIP, addresses #2247, replaces #2276 

Edit: command for running trec_eval:
```
java -jar target/trec_eval.jar qrels.txt runfile.trec
```

### CODE REVIEW

**Comment by jasper-xian** [comment]
Looking to mirror what's done in [pyserini](https://github.com/castorini/pyserini/blob/master/pyserini/eval/trec_eval.py), basically adding more checks for command line and runfile formatting.

Also looking to integrate built-in qrels from anserini-tools.

Let me know if this is what you had in mind @lintool

File: src/main/java/trec_eval.java [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+ * of platforms.
+ * <br><b>Supported Platforms</b><br>
+ * <ul>
+ * <li>Windows (Intel 32bit)</li>
+ * <li>Linux (Intel 32bit)</li>
+ * <li>Mac OS X (Intel 64bit)</li>
+ * </ul>
+ * @author Craig Macdonald
+ */
+public class trec_eval
```
Comment (lintool) [inline src/main/java/trec_eval.java:51]: Can you change to two-space indent, just like all our other code?
Opening braces go on previous line.

File: pom.xml [hunk 2]
```diff
@@ -279,6 +279,32 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-assembly-plugin</artifactId>
+        <version>2.6</version>
+        <configuration>
+					<archive>
```
Comment (lintool) [inline pom.xml:287]: fix indent?

File: src/main/java/trec_eval.java [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.LineIterator;
+
+/**
```
Comment (lintool) [inline src/main/java/trec_eval.java:40]: Can you explicitly acknowledge in comments that we copy/adapt from jtreceval?

File: pom.xml [hunk 4]
```diff
@@ -279,6 +279,32 @@
...
+						<manifest>
+							<mainClass>trec_eval</mainClass>
+						</manifest>
```
[the remark this answers is not in the document]
Comment (jasper-xian) [reply pom.xml]: hmm, I'm still picking up maven on the fly, but my assumption was to build a separate jar just with `trec_eval.java` for the purpose of only using trec_eval, if that makes sense.

so I thought having trec_eval as the main class for that jar `trec_eval.jar` made sense, let me know your thoughts

[the remark this answers is not in the document]
Comment (lintool) [reply pom.xml]: ah, that makes sense if you can build a separate jar.

**Comment by lintool** [comment]
Pulled branch, did a simple test on my iMac Pro (intel):

```
(pyserini-dev3) jimmylin@iMac-Pro anserini % tools/eval/trec_eval.9.0.4/trec_eval -m map -m P.30 tools/topics-and-qrels/qrels.adhoc.401-450.txt runs/run.disk45.robust04.bm25
map                   	all	0.2515
P_30                  	all	0.3560
(pyserini-dev3) jimmylin@iMac-Pro anserini % java -jar target/trec_eval.jar  -m map -m P.30 tools/topics-and-qrels/qrels.adhoc.401-450.txt runs/run.disk45.robust04.bm25
map                   	all	0.2515
P_30                  	all	0.3560
```

Works as expected... so far, so good!

**Comment by lintool** [comment]
I think we should remove all instances of the 9.0.4 binaries so we don't get confused?

**Comment by jasper-xian** [comment]
> I think we should remove all instances of the 9.0.4 binaries so we don't get confused?

Good idea, I've removed them

**Comment by lintool** [comment]
I've pulled the latest branch and ran the following tests:

https://github.com/castorini/anserini/blob/master/docs/regressions/regressions-disk45.md


```bash
nohup python src/main/python/run_regression.py --index --verify --search --regression disk45 >& logs/log.disk45 &

# old way
tools/eval/trec_eval.9.0.4/trec_eval -m map -m P.30 tools/topics-and-qrels/qrels.adhoc.401-450.txt runs/run.disk45.robust04.bm25
tools/eval/trec_eval.9.0.4/trec_eval -v

# new way (1)
java -jar target/trec_eval.jar -m map -m P.30 tools/topics-and-qrels/qrels.adhoc.401-450.txt runs/run.disk45.robust04.bm25
java -jar target/trec_eval.jar -v

# new way (2)
java -cp target/anserini-0.24.1-SNAPSHOT-fatjar.jar trec_eval -m map -m P.30 tools/topics-and-qrels/qrels.adhoc.401-450.txt runs/run.disk45.robust04.bm25
java -cp target/anserini-0.24.1-SNAPSHOT-fatjar.jar trec_eval -v
```

Everything works as expected on Linux (`linux.cs.uwaterloo.ca`), Mac Intel (my iMac Pro), and Mac ARM (my Mac Studio).

👍

**Pull request merged**